File: fastapi_lotto_result/app/utils/lotto_utils.py
import app.utils.date_utils as dateUtils
from datetime import datetime

_START_DATE = dateUtils.get_korean_date_from_ymd(2002, 12, 7)
_DRAW_HOUT_MIN = 2045
_SATURDAY = 5

def is_before_drawtime(date: datetime) -> bool:
    return date.hour*100 + date.minute < _DRAW_HOUT_MIN

def last_draw_no() -> int:
    date = dateUtils.get_korean_date()
    drw_no = dateUtils.week_diff(_START_DATE, date) + 1
    if (date.weekday != _SATURDAY or not is_before_drawtime(date)):
        return drw_no
    return drw_no - 1


def max_draw_no() -> int:
    date = dateUtils.get_korean_date()
    drw_no = dateUtils.week_diff(_START_DATE, date) + 1
    if (date.weekday != _SATURDAY ):
        return drw_no
    return drw_no + 1

def get_draw_date(drnNo : int) -> datetime:
    return dateUtils.add_weeks(_START_DATE, drnNo-1)

File: dotnet_winform_simpleLotto/simpleLotto/utils/LottoUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace simpleLotto.utils {
    public static class LottoUtils {
        private static DateTime START_DATE = DateUtils.GetUtcFromKoreanDateFromYmd(2002, 12, 7);
        private static int DRAW_HOUR_MIN = 2045;

        private static bool IsBeforeDrawTime(DateTime dateTime) {
            return (dateTime.Hour * 100 + dateTime.Minute) < DRAW_HOUR_MIN;
        }

        private static Tuple<DateTime, int> GetDrawInfo() {
            DateTime date = DateUtils.GetKoreanDateTime(DateTime.Now);
            int drwNo = DateUtils.WeekDiff(START_DATE, date) + 1;
            return Tuple.Create(date, drwNo);
        }
        public static int LastDrawnNo() {
            (DateTime date, int drwNo) = GetDrawInfo();
            if ( date.DayOfWeek != DayOfWeek.Saturday || !IsBeforeDrawTime(date)) {
                return drwNo;
            }
            return drwNo - 1;
        }

        public static int MaxDrawNo() {
            (DateTime date, int drwNo) = GetDrawInfo();
            return date.DayOfWeek != DayOfWeek.Saturday ? drwNo + 1 : drwNo;
        }

        public static DateTime getDrawDate(int drwNo) {
            return DateUtils.AddWeeks(START_DATE, drwNo-1);
        }
    }
}
